### ISSUE TITLE
[Feature Request] Add styles api for ActionIcon

### ISSUE THREAD

**Issue by 7iomka** [mantinedev/mantine#4282]
### What package has an issue

@mantine/core

### Describe the bug

Hi.
First of all - ActionIcon have the `button` behavior in terms of UX;
For now - Button component have the full styles api support, so, we can use in theme override on provider level `variants` field to declare some custom styles for existing variants.
After that, we need to make the similar variants override for ActionIcon, but that is not possible, because, how docs for v6 says: 

> You can now use MantineProvider to add variants to all Mantine components that support Styles API

So, this is a problem, because, for example, we have declared button variants:
```ts
type ComponentVariants =
  | Record<
      PropertyKey,
      (theme: MantineTheme, params: any, context: ContextStylesParams) => Record<string, CSSObject>
    >
  | undefined;

const buttonVariants: ComponentVariants = {
  filled: (theme, params: ButtonStylesParams) => {
    const colorProp = params.color || theme.primaryColor;

    return {
      root: {
        color:
          colorProp === 'secondary'
            ? '#fff' // secondary color by design force white text color
            : getReadableColor({
                color: getThemeColor({ theme, color: colorProp }),
              }),
      },
    };
  },
  outline: (theme, params) => {
    const resolvedColor = getThemeColor({ theme, color: params.color });
    const resolvedPassedColor = params.color ? resolvedColor : null;

    return {
      root: {
        color: resolvedPassedColor || 'inherit',
        borderColor: resolvedPassedColor || 'currentColor',
        '&:not([data-disabled])': theme.fn.hover({
          borderColor: resolvedColor,
          backgroundColor: theme.fn.rgba(
            getThemeColor({ theme, color: params.color, shade: 2 }),
            0.7,
          ),
        }),
      },
    };
  },
  subtle: (theme, params) => {
    const resolvedColor = getThemeColor({ theme, color: params.color });
    const resolvedPassedColor = params.color ? resolvedColor : null;
    const backgroundFallbackColor =
      theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[2];
    const backgroundColor = resolvedPassedColor
      ? theme.fn.rgba(resolvedPassedColor, 0.2)
      : backgroundFallbackColor;

    return {
      root: {
        color: resolvedPassedColor ?? 'var(--root-color)',
        '&:not([data-disabled])': theme.fn.hover({
          color: resolvedPassedColor ?? 'var(--root-color)',
          backgroundColor,
        }),
      },
    };
  },
};

```

And we want to reuse them:
```

const actionIconVariants: ComponentVariants = {
  ...buttonVariants,
};
```

If you can add styles api support for ActionIcon, code from above will work fine.

### What version of @mantine/hooks page do you have in package.json?

latest

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
Styles API and variants api will be improved in 7.0. The changes to the Styles API will make it possible to do everything described in this issue + all of your previous questions and issues are also covered. 

You can find more information here:
- https://v7.mantine.dev/styles/styles-api
- https://v7.mantine.dev/styles/variants-sizes

**Comment by 7iomka** [comment]
> 

Version 7 looks like a completely different library fundamentally because of too many breaking changes.
Especially it will come out sometime this year.
You won't ignore such suggestions/problems in favor of the fact that it will be implemented in version 7 someday, will you?

**Comment by rtivital** [comment]
It is already implemented in 7.0. I do not have much capacity to focus on both versions, so there won't be much 6.x releases.
Regarding your initial question – ActionIcon supports styles api, you can apply the same variants as in Button component – https://codesandbox.io/s/broken-morning-42givk?file=/src/App.tsx

**Comment by rtivital** [comment]
ActionIcon has default color gray, Button does not have it

**Comment by 7iomka** [comment]
> ActionIcon has default color gray, Button does not have it

this is a problem because the default value is not expected to be in `props.color`. 
But even if you tell me this is expected behavior and tell me why, I still need to know if the user passed on the color - how do I do that? Because I'm stumped.

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.
